src/network/jsonwebtask.h

```diff
@@ -25,36 +25,53 @@ struct JsonWebRequest final {
     QJsonDocument content;
 };
 
-struct JsonWebResponse : public WebResponse {
+class JsonWebResponse final {
   public:
     static void registerMetaType();
 
     JsonWebResponse() = default;
     JsonWebResponse(
-            WebResponse response,
-            QJsonDocument content)
-            : WebResponse(std::move(response)),
-              content(std::move(content)) {
+            WebResponse&& response,
+            QJsonDocument&& content)
+            : m_response(std::move(response)),
+              m_content(std::move(content)) {
```

Comment:
std::move is redundant if we all read have the l-value reference. 
However I think the compiler is smart enough to produce optimal code if we use a copy by value parameter and then call the function using std::move.
Is this true?

Reply:
The r-value references enforce that the components are actually moved into the wrapper. This is the intended semantic here.

Reply:
Than the std::move(response) and std::move(content) is redundant.